moxie-protocol/src/subject-token.ts

```diff
@@ -33,23 +33,19 @@ export function handleTransfer(event: Transfer): void {
   // updating portfolios
   if (!mint) {
     let fromAddressPortfolio = getOrCreatePortfolio(from, contractAddress, event.transaction.hash, event.block)
-    let updatedBalance = fromAddressPortfolio.balance.minus(value)
-    if (updatedBalance.equals(BigInt.fromI32(0))) {
-      subjectToken.uniqueHolders = subjectToken.uniqueHolders.minus(BigInt.fromI32(1))
-      store.remove("Portfolio", fromAddressPortfolio.id)
-    } else {
-      fromAddressPortfolio.balance = updatedBalance
-      savePortfolio(fromAddressPortfolio, event.block)
-    }
+    fromAddressPortfolio.unstakedBalance = fromAddressPortfolio.unstakedBalance.minus(value)
+    savePortfolio(fromAddressPortfolio, event.block,true)
   }
   if (!burn) {
     let toAddressPortfolio = getOrCreatePortfolio(to, contractAddress, event.transaction.hash, event.block)
     // adding unique holders when a new portfolio is created
-    if (toAddressPortfolio.balance.equals(BigInt.fromI32(0))) {
-      subjectToken.uniqueHolders = subjectToken.uniqueHolders.plus(BigInt.fromI32(1))
+    if (toAddressPortfolio.unstakedBalance.equals(BigInt.fromI32(0))) {
```

Comment:
This logic will fail if transfer for zero amount will happen.

You should first update balance then then do 0 balance check.

Reply:
moved this logic to `getOrCreatePortfolio` , so whenever new portfolio gets created, uniqueHolders increment

and in `savePortfolio` we decrement and delete portfolio if balance is zero